File: java/src/main/java/com/tiagoamp/gameoflife/model/UniverseBuilder.java
package com.tiagoamp.gameoflife.model;

import java.util.Random;
import java.util.function.Supplier;

public class UniverseBuilder {

    public static Universe from(int[][] intSeed) {
        int length = intSeed.length;
        State[][] seed = new State[length][length];
        for (int r = 0; r < length; r++) {
            for (int c = 0; c < length; c++) {
                seed[r][c] = State.fromValue(intSeed[r][c]);
            }
        }
        return new Universe(seed);
    }

    public static Universe fromRandomState(int gridSize) {
        State[][] seed = new State[gridSize][gridSize];
        for (int r = 0; r < gridSize; r++) {
            for (int c = 0; c < gridSize; c++) {
                seed[r][c] = State.fromValue( eitherZeroOrOne.get() );
            }
        }
        return new Universe(seed);
    }


    private static Supplier<Integer> eitherZeroOrOne = () -> new Random().nextInt(2);

}

File: javascript/UniverseBuilder.js
const Universe = require('./Universe');

class UniverseBuilder {

    static fromRandomState(gridSize) {
        const seed = [];
        for (let r = 0; r < gridSize; r++) {
            const row = [];
            for (let c = 0; c < gridSize; c++) {
                row[c] = eitherZeroOrOne();
            }
            seed.push(row);
        }
        return new Universe(seed);
    }

}

eitherZeroOrOne = () => {
    const min=0, max=1;  
    return Math.floor(Math.random()*(max-min+1)+min);
}

module.exports = UniverseBuilder;